File: lib/lab1/letter_writer.rb
puts "Enter your first name:"
name = gets.chomp
puts "Enter the name of your friend:"
friend_name = gets.chomp
puts "What is your friend's gender?"
gender = gets.chomp
pronoun = gender == 'm' ? "him" : "her"
puts "What is your friend's age?"
age = gets.chomp.to_i
age_comment = case
when age < 12
  "Next year you will be #{age+1}."
when age == 17
  "Next year you will be able to vote."
when age > 70
  "I hope you are enjoying retirement."
end

puts "Dear #{name},"
puts "\tHow are you? How have you been? I am fine. I miss you. Have you seen #{friend_name} recently? If you see #{friend_name} tell #{pronoun} to call me. I heard you just had a birthday and you're #{age} years old. #{age_comment}"
puts "Yours sincerely,"
puts "\n\nWill C."

File: lib/lab1/letter_writer.cpp
#include <iostream>
#include "../std_lib_facilities.h"

using namespace std;

int main()
{
  string name, friend_name, pronoun;
  char gender;
  int age;
  
  cout << "Enter your first name:\n";
  cin >> name;
  cout << "Enter the name of your friend:\n";
  cin >> friend_name;
  cout << "What is your friend's gender?\n";
  cin >> gender;
  if(gender == 'm'){
    pronoun = "him";
  } else {
    pronoun = "her";
  }
  cout << "What is your friend's age?\n";
  cin >> age;
  if(age <= 0 || age >= 110){
    simple_error("You're kidding!");
  }
  cout << "Dear " << name << ",\n";
  cout << "\tHow are you?  I am fine.  I miss you.  ";
  cout << "Have you seen " << friend_name << " recently?  ";
  cout << "If you see " << friend_name << " tell " << pronoun << " to call me.  ";
  cout << "I heard you just had a birthday and you're " << age << " years old now.  ";
  if(age < 12){
    cout << "Next year you will be " << (age + 1) << ".  ";
  } else if(age == 17){
    cout << "Next year you will be able to vote.  ";
  } else if(age > 70){
    cout << "I hope you are enjoying retirement.  ";
  }
  cout << "\nYours sincerely,\n\n" << name << " ";
  return 0;
}
